instrumentation/dubbo/src/main/java/brave/dubbo/DubboParser.java

```diff
@@ -25,6 +29,36 @@
 import org.apache.dubbo.rpc.support.RpcUtils;
 
 final class DubboParser {
+  static final Map<Integer, String> ERROR_CODE_NUMBER_TO_NAME = errorCodeNumberToName();
+
+  /**
+   * Dubbo adds error codes sometimes. For example, they added two in the last two years. If we did
+   * a simple switch/case of each code, like {@link RpcException#BIZ_EXCEPTION}, new code names
+   * would not be seen until an upgrade of Brave. That, or we'd have to fall back to a code number
+   * until the upgrade of Brave converged.
+   *
+   * <p>This uses reflection instead, which is unlikely to fail as Dubbo's error codes are public
+   * constants.
+   */
+  static Map<Integer, String> errorCodeNumberToName() {
+    Map<Integer, String> result = new LinkedHashMap<>();
+    for (Field field : RpcException.class.getDeclaredFields()) {
+      if (Modifier.isStatic(field.getModifiers()) && field.getType() == int.class) {
```

Comment:
How about checking `Modifier.isPublic` too?

Reply:
yeah good idea. it makes more sense reading it now, also.